src/field.rs

```diff
@@ -23,6 +23,14 @@ impl PlutoField {
   pub const ORDER_U32: u32 = PLUTO_FIELD_PRIME;
 
   pub fn new(value: u32) -> Self { Self { value } }
+
+  pub fn primitive_root_of_unity(n: u32) -> Self {
```

Comment:
can you comment why this works and what this does? i'm not familiar with what you've done here